cmd/tether/attach.go

```diff
@@ -140,78 +171,121 @@ func (t *attachServerSSH) start() error {
 	}
 	t.sshConfig.AddHostKey(pkey)
 
-	t.enabled = true
+	t.Enabled()
 	go t.run()
 
 	return nil
 }
 
 // stop is not thread safe with start
-func (t *attachServerSSH) stop() {
+func (t *attachServerSSH) stop() error {
 	defer trace.End(trace.Begin("stop attach server"))
 
-	if t == nil || !t.enabled {
-		return
+	if t == nil {
+		err := fmt.Errorf("attach server is not configured")
+		log.Error(err)
+		return err
+	}
+
+	if !t.IsEnabled() {
+		err := fmt.Errorf("attach server is not enabled")
+		log.Error(err)
+		return err
 	}
 
-	t.enabled = false
-	conn := t.conn
-	t.conn = nil
+	log.Debugf("Setting enabled to false")
+	t.Disabled()
 
-	if conn != nil {
-		(*conn).Close()
+	log.Debugf("Canceling the context")
+	t.cancel()
+
+	log.Debugf("Acquiring the connection lock")
+	t.conn.Lock()
+	if t.conn.conn != nil {
```

Comment:
Looks like in `Start()`, Enabled is set, then `t.conn.Lock()` is acquired, and `t.conn.conn` is assigned in `run()`.  But here, we set disabled, acquire the `conn` lock , then do this check.  Is it possible Start() will set enabled, then in another routine we call `stop`, set disabled, acquire the `conn.Lock`, find this nil before `run()` has a chance to assign, fall out of `stop`, and run now has a dangling `conn`?